flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManagerDriver.java

```diff
@@ -359,8 +363,17 @@ public void onError(List<KubernetesPod> pods) {
         }
 
         @Override
-        public void handleFatalError(Throwable throwable) {
-            getResourceEventHandler().onError(throwable);
+        public void handleError(Throwable throwable) {
+            if (throwable instanceof KubernetesTooOldResourceVersionException) {
+                getMainThreadExecutor()
+                        .execute(
+                                () -> {
+                                    log.info("Creating a new watch on TaskManager pods.");
+                                    podsWatchOpt = watchTaskManagerPods();
```

Comment:
I think we need to check whether the `KubernetesResourceManagerDriver` is still running or has been `terminated` already. Moreover, we should probably close the old `podsWatchOpt` to release potentially acquired resources.

Reply:
We have done the running check in `KubernetesLeaderElectionDriver` and `KubernetesLeaderRetrievalDriver`. The reason why I have not done it here is that the JobManager process will exit if `KubernetesResourceManagerDriver#terminate` is executed. After then all the resources(e.g websockets, executors) will be destroyed. It seems that we do not need to do the clean up. I could add it back if you think it is really necessary.

For the old `podsWatchOpt`, I think it does not need to be released because `WatchCallbackHandler#handleError` will only be called when the watcher is already closing with exception. So I think we do not need to do the duplicated close.

Reply:
It might be better if we could release the resources(e.g. websocket, executors) before the JobManager process exits.